### PR TITLE
Podcast page

### LINKED ISSUE #200: Design podcast page as attached
newsdesk-select
Screen Shot 2022-03-29 at 17 27 44
1643630496828
![1644254577108-7dc2915c39838d8b9208db03ed77cb56
-1bc3d005c2ca5637ae9e25f0064f4ca2](https://user-images.githubusercontent.com/104105362/168733204-06c119c5-d033-474f-b470-0950bde45d91.jpg)

1650876885932-84b6297f1bbd36a5f2090124ba92b738
Screen Shot 2022-03-29 at 17 27 44
1650876885932-84b6297f1bbd36a5f2090124ba92b738
1643630496828-1bc3d005c2ca5637ae9e25f0064f4ca2
1644254577108-7dc2915c39838d8b9208db03ed77cb56
1646671323029-6b5eaf38935919cc646c005eed2b17ea
1644254294959-d985c5ef35bd9770d72f5b3f439acf6e
1636625196585-69fd4b45813ee5032c2e5a80ec647b68
IMG_7667

### PR DESCRIPTION
**Pull request by bhavinsen**
Fixed #200 and #109

### CODE REVIEW

**Comment by mrjamieeast** [comment]
please resize thumbnails and position as per the design

**Comment by mrjamieeast** [comment]
We do not need the drop downs on this page (select podcast and date)

**Comment by mrjamieeast** [comment]
This works thanks

File: cypress/integration/podcasts.spec.js [hunk 1]
```diff
@@ -16,8 +17,6 @@ describe("newsdesk logged in", () => {
     cy.visit(homeurl)
     cy.get("#login").click()
     cy.contains("Peach Otter").click()
-    cy.get("h2").should("have.text", "⬆️ select a podcast ⬆️")
-    switchPodCast("dev sandbox")
```
Comment (chrisns) [inline cypress/integration/podcasts.spec.js:20]: the context of all these tests is within a podcast, so you may as well leave the switch in

File: cypress/integration/podcasts.spec.js [hunk 2]
```diff
@@ -36,34 +35,32 @@ describe("newsdesk logged in", () => {
   })
 
   it("should maximise the inbox", () => {
+    cy.get("#select-podcast").eq(0).click()
```
Comment (chrisns) [inline cypress/integration/podcasts.spec.js:38]: either use the switchPodCast or leave this in the beforeEach

File: cypress/integration/podcasts.spec.js [hunk 3]
```diff
@@ -2,7 +2,8 @@
 
 const homeurl = "http://localhost:3000"
 
-const switchPodCast = (podcast) => {
+const switchPodCast = (podcast, add) => {
+  if (add == true) cy.get("#select-podcast").eq(0).click()
```
Comment (chrisns) [inline cypress/integration/podcasts.spec.js:6]: not sure what add really means here, its not adding anything, maybe a better var name?
also if you're going to do an if it should be `===`

File: src/components/organisms/AppBar.vue [hunk 4]
```diff
@@ -104,26 +120,26 @@
...
 import { ArrowCircleDownIcon,ArrowCircleUpIcon, MoonIcon, SunIcon } from "@heroicons/vue/outline"
 import { usePodcastStore } from "@/store/podcasts"
+import { useDark, useToggle } from "@vueuse/core"
 
 const authStore = useAuthStore()
 const route = useRoute()
 const router = useRouter()
 const store = usePodcastStore()
-
+const date = ref(new Date().toISOString().substr(0, 10))
```
Comment (chrisns) [inline src/components/organisms/AppBar.vue:135]: date needs to default to tomorrow

**Review by chrisns — CHANGES_REQUESTED** [review]

**Review by chrisns — APPROVED** [review]

File: src/components/organisms/AppBar.vue (hunk 4 above)
Comment (chrisns) [reply src/components/organisms/AppBar.vue]: this still needs to be tomorrow

**Review by chrisns — CHANGES_REQUESTED** [review]

File: src/components/organisms/AppBar.vue (hunk 4 above)
Comment (bhavinsen) [reply src/components/organisms/AppBar.vue]: I have already fixed it.

Please refer -> [https://i.imgur.com/fPV9DzV.png](url)
                      [https://github.com/DaftDoris/newsdesk/pull/229/commits/16171770e650dff9684c17d4bc8421d0448c5d07](url)

File: src/components/organisms/AppBar.vue [hunk 5]
```diff
@@ -104,26 +120,27 @@
...
 import { usePodcastStore } from "@/store/podcasts"
+import { useDark, useToggle } from "@vueuse/core"
 
 const authStore = useAuthStore()
 const route = useRoute()
 const router = useRouter()
 const store = usePodcastStore()
-
+const todayDate = new Date(new Date().getTime() + 24 * 60 * 60 * 1000)
+const date = ref(todayDate.toISOString().substr(0, 10))
```
Comment (chrisns) [inline src/components/organisms/AppBar.vue:136]: 1. you're calling the variable todayDate but its actually tomorrow
2. defining a date by adding some milliseconds seconds is ugly and theres weird edge cases where this won't work
3. theres no inherent need to store timestamps in the db, it'd be way more readable and easier to debug if you instead used yyyymmdd
4. substringing a number is dangerous, safer would be divide by a number and `math.floor` it

could/should be made a lot easier to read by splitting over a few lines and avoiding so many `new Date` but this is safer: `new Date(new Date().setDate((new Date()).getDate() + 1)).toISOString().split('T')[0]`

**Review by chrisns — CHANGES_REQUESTED** [review]

**Comment by chrisns** [comment]
Lgtm. @mrjamieeast Please check this works

**Comment by mrjamieeast** [comment]
Yup great

**Pull request merged**